sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -299,6 +300,34 @@ public Instant getCurrentTimestamp() throws NoSuchElementException {
     return curTimestamp;
   }
 
+  private static final byte[] EMPTY_RECORD_ID = new byte[0];
+
+  @Override
+  public byte[] getCurrentRecordId() throws NoSuchElementException {
+    if (!this.offsetBasedDeduplicationSupported) {
+      // BoundedReadFromUnboundedSource and tests may call getCurrentRecordId(), even for non offset
+      // deduplication cases. Therefore, Kafka reader cannot produce an exception when offset
+      // deduplication is disabled. Instead an empty record ID is provided.
+      return EMPTY_RECORD_ID;
+    }
+    if (curRecord != null) {
```

Comment:
nit: how about checking == null here and throwing. the exception clearly tied to the check then

Reply:
Sg, I had switched things around when confused about some nullable lint check.